### PR TITLE
PyTorch support for gradient-based weights for mixed-precision metric

### PR DESCRIPTION
**Pull request by ofirgo**
* Added support to gradient-based weights for the distance metric computation in Pytorch framework.
* Default value for alpha parameter (tuning between output layers and regular layers) is changed to 0.3 (based on experimental study).
* Default weights for metric computation remained uniform (gradient-based weights option needs to be activated by an argument to mixed-precision quantization config).
* Minor fix in Keras gradient model tests TPC imports.

### CODE REVIEW

File: model_compression_toolkit/core/pytorch/back2framework/model_gradients.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+            op_func = n.type if isinstance(n, FunctionalNode) else getattr(self, n.name)
+            out_tensors_of_n = run_operation(n,  # Run node operation and fetch outputs
+                                             input_tensors,
+                                             op_func=op_func)
+
+            if isinstance(out_tensors_of_n, list):
+                output_t = []
+                for t in out_tensors_of_n:
+                    t.retain_grad()
```
Comment (reuvenperetz) [inline model_compression_toolkit/core/pytorch/back2framework/model_gradients.py:153]: 1. Why we don't check here if the tensor requires_grad?
2. Shouldn't we use gradients only for interest points?
